packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,547 @@
+    } else {
+      childWidth = constraints.minWidth / childCount;
+      for(RenderBox child in getChildrenAsList()) {
+        childWidth = max(childWidth, child.getMaxIntrinsicWidth(double.infinity));
+      }
+    }
+
+    RenderBox child = firstChild;
+    while (child != null) {
+      final _SegmentedControlContainerBoxParentData childParentData = child.parentData;
+      final double boxHeight = child.getMaxIntrinsicHeight(childWidth);
+      maxHeight = max(maxHeight, boxHeight);
+      child = childParentData.nextSibling;
+    }
+
+    constraints.constrainHeight(maxHeight);
+
+    final BoxConstraints childConstraints = new BoxConstraints.tightFor(
+      width: childWidth,
+      height: maxHeight,
+    );
+
+    child = firstChild;
+    while (child != null) {
+      final _SegmentedControlContainerBoxParentData childParentData = child.parentData;
+      child.layout(childConstraints, parentUsesSize: true);
+      child = childParentData.nextSibling;
+    }
+
+    switch (textDirection) {
+      case TextDirection.rtl:
+        RenderBox child = lastChild;
+        _layoutRects(
+          () {
+            final _SegmentedControlContainerBoxParentData childParentData = child.parentData;
+            child = childParentData.previousSibling;
+            return childParentData.previousSibling;
+          },
+          lastChild,
+          firstChild,
+        );
+        break;
+      case TextDirection.ltr:
+        RenderBox child = firstChild;
+        _layoutRects(
+          () {
+            final _SegmentedControlContainerBoxParentData childParentData = child.parentData;
+            child = childParentData.nextSibling;
+            return childParentData.nextSibling;
+          },
+          firstChild,
+          lastChild,
+        );
+        break;
+    }
+
+    size = constraints.constrain(new Size(childWidth * childCount, maxHeight));
+  }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    fillPaint.clear();
+
+    RenderBox child = firstChild;
+    int index = 0;
+    while (child != null) {
+      final _SegmentedControlContainerBoxParentData childParentData = child.parentData;
+      _paintChild(context, offset, child, index);
+      child = childParentData.nextSibling;
+      index += 1;
+    }
+  }
+
+  void _paintChild(PaintingContext context, Offset offset, RenderBox child, int childIndex) {
+    assert(child != null);
+
+    final _SegmentedControlContainerBoxParentData childParentData = child.parentData;
+
+    Color color = CupertinoColors.white;
+    if (selectedIndex != null && selectedIndex == childIndex) {
+      color = CupertinoColors.activeBlue;
+    } else if (pressedIndex != null && pressedIndex == childIndex) {
+      color = _kPressedBackground;
+    }
+
+    fillPaint.add(
+      new Paint()
+        ..color = color
+        ..style = PaintingStyle.fill,
+    );
+
+    context.canvas.drawRRect(
+      childParentData.surroundingRect.shift(offset),
+      fillPaint.last,
```

Comment:
you don't actually use the fillPaint array any more. You could simplify the code by removing fillPaint entirely, and just having a local Paint variable here.

Reply:
We use fillPaint in our tests, because we don't have another way to get the background color of each option - is it okay to keep it for that purpose?